### PR TITLE
PLATFORM-366 Fix mypy structural errors for library

### PR DESCRIPTION
**Pull request by simonwgill**
## Describe your changes

- Adds minimum required structural files to allow this to be a typed library
- Adds mypy to venv
- Performs minimal fixes to resolve type errors.
- Adds informational type checking action

Testing using a git reference to this branch and importing into eks-platform-cdk. This removes the errors from the library being structured badly.

The change will all the null coalescing could have been done another way. The problem was that the argument was being passed directly from a function with optional arguments and a default to a function which required a non-null argument. This change doesn't affect the interface at all, so it's the one I've gone with, but one of the other alternatives of changing the function definitions might be the way forward in the long run..

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] I have updated the Readme / docs
- [X] If this is a breaking change, I have discussed the roll-out and roll-back plan with the team(s) and I have provided detailed instructions
- [X] If it is a core feature, I have added thorough tests

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,3 +31,9 @@ select = ['E', 'W', 'F', 'C', 'N']
 ignore = ['E501']
 line-length = 95
 exclude = ["ca_cdk_constructs/eks/imports"]
+
+[[tool.mypy.overrides]]
```
Comment (michelesr) [inline pyproject.toml:35]: why the double square brackets?

Comment (simonwgill) [reply pyproject.toml:35]: Good question, that was the example from mypy. Not sure what it's doing in pyproject syntax, but I'll have to have a look

Comment (simonwgill) [reply pyproject.toml:35]: It's the [TOML table array syntax](https://toml.io/en/v1.0.0#array-of-tables). It means you can have multiple `[[tool.mypy.overrides]]` elements, each one setting a different config for different packages.

**Review by michelesr — APPROVED** [review]
LGTM

**Review by zhelyan — APPROVED** [review]
Lgtm, sorry it took so long, I just saw it

**Pull request merged**